src/main/java/dev/waterdog/waterdogpe/network/downstream/SwitchDownstreamHandler.java

```diff
@@ -48,6 +49,34 @@ public DownstreamSession getDownstream() {
         return this.client.getSession();
     }
 
+    @Override
+    public final boolean handle(SetDisplayObjectivePacket packet) {
+        this.player.getScoreboards().add(packet.getObjectiveId());
+        return false;
+    }
+
+    @Override
+    public final boolean handle(RemoveObjectivePacket packet) {
+        this.player.getScoreboards().remove(packet.getObjectiveId());
+        return false;
+    }
+
+    @Override
+    public final boolean handle(SetScorePacket packet) {
+        for(ScoreInfo info : packet.getInfos()) {
+            this.player.getScoreInfos().stream().
+                    filter(info1 -> info1.getScoreboardId() == info.getScoreboardId()).
+                    findFirst().ifPresent(info1 -> this.player.getScoreInfos().remove(info1));
+        }
+
+        if(packet.getAction() == SetScorePacket.Action.SET) {
+            for(ScoreInfo info : packet.getInfos()) {
+                this.player.getScoreInfos().add(info);
+            }
+        }
+        return false;
+    }
+
```

Comment:
Duplicated code because servers may be sending the scoreboards too early which causes the proxy to not add them to the ProxiedPlayer, this fixes it for now.